functions/supplier/syslog-supplier/src/main/java/org/springframework/cloud/fn/supplier/syslog/SyslogSupplierProperties.java

```diff
@@ -125,6 +125,23 @@ public boolean isSupportedRfc() {
 
 	@AssertTrue(message = "protocol must be 'udp' or 'tcp' or 'both'")
 	public boolean isSupportedProtocol() {
-		return "tcp".equals(this.protocol) || "udp".equals(this.protocol) || "both".equals(this.protocol);
+		return "tcp".equals(this.protocol.name()) || "udp".equals(this.protocol.name()) || "both".equals(this.protocol.name());
```

Comment:
I think this check is pointless: you just can't provide anything else unless a constant from `Protocol`.